service/static/js/rest_api.js

```diff
@@ -393,5 +435,176 @@ $(function () {
+    $("#addr-create-btn").click(function () {
+
+        var customer_id = $("#cust_id").val();
+        var street_address = $("#addr_street_address").val();
+        var city = $("#addr_city").val();
+        var state = $("#addr_state").val();
+        var country = $("#addr_country").val();
+        var zipcode = $("#addr_zip").val();
+
+        var data = {
+            "customer_id": customer_id,
+            "street_address": street_address,
+            "city": city,
+            "state": state,
+            "country": country,
+            "zipcode": zipcode
+        };
+
+        var ajax = $.ajax({
+            type: "POST",
+            url: "/customers/" + customer_id+"/addresses",
+            contentType: "application/json",
+            data: JSON.stringify(data),
+        });
+
+        ajax.done(function(res){
+            update_customer_form_data(res)
+            update_address_form_data(res)
+            flash_message("Success")
+        });
+
+        ajax.fail(function(res){
+            flash_message(res.responseJSON.message)
+        });
+    });
+
+    // ****************************************
+    // Update a Address
+    // ****************************************
+    $("#addr-update-btn").click(function () {
+        var customer_id = $("#cust_id").val();
+        var address_id = $("#addr_id").val();
+        
+        var street_address = $("#addr_street_address").val();
+        var city = $("#addr_city").val();
+        var state = $("#addr_state").val();
+        var country = $("#addr_country").val();
+        var zipcode = $("#addr_zip").val();
+
+        var data = {
+            "street_address": street_address,
+            "city":city,
+            "state": state,
+            "zipcode": zipcode,
+            "country": country
+        };
+
+        var ajax = $.ajax({
+                type: "PUT",
+                url: "/customers/" + customer_id+"/addresses/" +address_id,
+                contentType: "application/json",
+                data: JSON.stringify(data)
+            })
+
+        ajax.done(function(res){
+            res.id = customer_id
+            update_address_form_data(res)
+            flash_message("Success")
+        });
+
+        ajax.fail(function(res){
+            flash_message(res.responseJSON.message)
+        });
+
+    });
+
+    // ****************************************
+    // Retrieve an Address
+    // ****************************************
+    $("#addr-search-btn").click(function () {
+
+        var customer_id = $("#cust_id").val();
+        var address_id = $("#addr_id").val();
+
+        var ajax = $.ajax({
+            type: "GET",
+            url: "/customers/" + customer_id+"/addresses/" +address_id,
+            contentType: "application/json",
+        })
+
+        ajax.done(function(res){
+            clear_form_data()
+            add_single_address(res)
+            update_address_form_data(res)
+            flash_message("Success")
```

Comment:
I think it might be easier user-wise if the retrieved address is shown on the actual form instead of the retrieved address results. because the retrieved address results isn't visible on my browser until i scroll down, it almost looks like this button doesn't work until I scroll down

Reply:
> I think it might be easier user-wise if the retrieved address is shown on the actual form instead of the retrieved address results. because the retrieved address results isn't visible on my browser until i scroll down, it almost looks like this button doesn't work until I scroll down

Makes sense.